firebase-perf/src/main/java/com/google/firebase/perf/gauges/CpuGaugeCollector.java

```diff
@@ -96,6 +97,7 @@ private CpuGaugeCollector() {
   }
 
   @VisibleForTesting
+  @Inject
```

Comment:
It doesn't allow me to add a comment on the unchanged code so adding it here.

Why are we still keeping the private constructor and the `sharedInstance` reference?

*same for memory collector class.

Reply:
We are adding DI step by step. First, make all classes constructor DI compatible. Second, remove all shareInstance and getInstance(). Let me know if you have questions.

Reply:
I think we chose bottom up approach to avoid coming back to these classes again. Do you think it would not be feasible to have those `sharedInstance` and `getInstance()` replaced now? And if yes, what's blocking us?

fyi: even I need to answer this question for the GaugeMetadaManager class changes (_which I am on the way_)

Reply:
Looks like Dagger created this class called `GaugeMetadataManager_Factory` for `GaugeMetadataManager` with public static methods for me to create new instance. 

I think if you mark your class as `@Singleton` you should also be able to get the singleton instance and hence won't need to have your own `getInstance()` method.

Reply:
1. Do you think it would not be feasible to have those sharedInstance and getInstance() replaced now?
Not really. For example, GaugeManager calls CpuGaugeManager.getInstance() and we cannot remove `getInstance()` unless make GaugeManager DI compatible.

2.I think if you mark your class as `@Singleton` you should also be able to get the singleton instance and hence won't need to have your own `getInstance()` method.
You are right @Singleton will make the class singleton. However, for the same reason above, we cannot remove `getInstance()` method now. Once we have all classes DI compatible, we can easily remove `getInstance()` usages in its caller and inject the class into the caller's constructors.

Reply:
Do I understand correctly that this class will be used by both DI injection(from other classes) as well as with getInstance() methods? if so, be careful as you may end up with multiple instances of this class, one created and maintained by dagger as a singleton, and one that you maintain youself in getInstance()

Reply:
The idea is to make the class DI-compatible, while maintaining the getInstance() method to avoid having a huge PR. While getInstance() exists, all users of the class should be using that, and then, in a single PR, we should remove the getInstance() method and use the injected instance instead.

**It is important to do the step in one go once we migrate because of what @vkryachko is saying - we won't actually have a singleton if we update usages separately.**

Reply:
Isn't there a way for us to get the singleton instance created by Dagger?

I know with Constructor injection we don't even need `getInstance()` but I was wondering to remove the `getInstance()` from the `CpuGaugeCollector` and replace the usages of `CpuGaugeCollector.getInstance()` with `DaggerGeneratedCpuGaugeCollector.getInstance()` (_or something along the lines_)?

Reply:
There is no way to remove `getInstance` now unless making the caller DI compatible. As Julio mentioned, this PR is our first step as making the current classes DI compatible without remove `getInstance`, which has the singleton problem as mentioned above and not in a releasable state. Therefore, we are merging to the feature branch called `perf-dagger`.